File: core/autonomy/semantic_constraint_solver.py
from __future__ import annotations

from typing import Any, Dict, List


def solve_semantic_constraints(
    constraints: List[Dict[str, Any]],
) -> Dict[str, Any]:

    valid = []

    invalid = []

    for constraint in constraints:

        if constraint.get("valid", True):
            valid.append(constraint)
        else:
            invalid.append(constraint)

    return {
        "valid_constraints": valid,
        "invalid_constraints": invalid,
    }

File: src/autonomy/semanticConstraintSolver.ts
/**
 * Converted from Python: core/autonomy/semantic_constraint_solver.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function solveSemanticConstraints(constraints: any): any {
  var valid: any[] = [];
  var invalid: any[] = [];
  var constraint: any;
  for (constraint of py.iter(constraints)) {
    if (py.truthy(py.get(constraint, "valid", true))) {
      py.listAppend(valid, constraint);
    } else {
      py.listAppend(invalid, constraint);
    }
  }
  return {"valid_constraints": valid, "invalid_constraints": invalid};
}
